Analysis/DNN_Application.py

```diff
@@ -0,0 +1,156 @@
+        ]
+        self.models = self._load_models()
+
+    ### Init helpers ###
+
+    def _set_environ_vars(self):
+        sys.path.append(os.environ["ANALYSIS_PATH"])
+        ana_path = os.environ["ANALYSIS_PATH"]
+        for header in [
+            "FLAF/include/Utilities.h",
+            "include/Helper.h",
+            "include/HmumuCore.h",
+            "FLAF/include/AnalysisTools.h",
+            "FLAF/include/AnalysisMath.h",
+        ]:
+            DeclareHeader(os.environ["ANALYSIS_PATH"] + "/" + header)
+
+    def _load_global_config(self):
+        filepath = os.path.join(os.environ["ANALYSIS_PATH"], "config", "global.yaml")
+        with open(filepath, "r") as f:
+            global_config = yaml.safe_load(f)
+        return global_config
+
+    def _load_models(self):
+        """
+        Load in the trained ONNX models
+        """
+        directory = os.path.join(os.environ["ANALYSIS_PATH"], "Analysis", "models")
+        models = []
+        renorm_vars = []
+        for i in range(self.parity):
+            # The model itself
+            filename = os.path.join(directory, f"trained_model_{i}.onnx")
+            model = ort.InferenceSession(filename)
+            models.append(model)
+        return models
+
+    def _load_dnn_config(self):
+        """
+        Read in the config used to train the network and input features used
+        """
+        directory = os.path.join(os.environ["ANALYSIS_PATH"], "Studies", "DNN")
+        filepath = os.path.join(directory, "configs", "config.toml")
+        with open(filepath, "rb") as f:
+            config = tomllib.load(f)
+        pairity = config["kfold"]["k"]
+        # Input features
+        filepath = os.path.join(directory, "ds_setup", "general.yaml")
+        with open(filepath, "r") as f:
+            columns_config = yaml.safe_load(f)
+        input_features = parse_column_names(
+            columns_config["vars_to_save"], column_type="data"
+        )
+        return pairity, input_features
+
+    def _load_framework(self):
+        """
+        Load any needed files from FLAF
+        """
+        sys.path.append(os.environ["ANALYSIS_PATH"])
+        ROOT.gROOT.ProcessLine(".include " + os.environ["ANALYSIS_PATH"])
+        ROOT.gInterpreter.Declare(f'#include "FLAF/include/Utilities.h"')
+        ROOT.gROOT.ProcessLine(f'#include "FLAF/include/HistHelper.h"')
+        ROOT.gROOT.ProcessLine(f'#include "FLAF/include/AnalysisTools.h"')
+        ROOT.gROOT.ProcessLine(f'#include "FLAF/include/AnalysisMath.h"')
+        ROOT.gROOT.ProcessLine(f'#include "FLAF/include/MT2.h"')
+        ROOT.gROOT.ProcessLine(f'#include "FLAF/include/Lester_mt2_bisect.cpp"')
+
+    ### Functions for running the inference ###
+
+    def prepare_dfw(self, dfw):
+        print("*********** Running prepare_dfw...")
+        dfw = analysis.DataFrameBuilderForHistograms(
+            dfw.df, self.global_cfg, self.period
+        )
+        dfw = analysis.PrepareDfForHistograms(dfw)
```

Comment:
Do you use **exactly** the same functions that are included in PrepareDfForHistograms? because in case we need to change the name from `PrepareDfForHistograms`to something more general. If you use only a subset of the functions defined in `PrepareDfForHistograms` you can modify `PrepareDfForNN` and include only the functions you actually need.